res/controllers/Denon-MC7000-scripts.js

```diff
@@ -1008,31 +1228,81 @@ MC7000.HotCueLED = function(value, group) {
 
 // Sampler LED
 MC7000.SamplerLED = function() {
-    for (var j = 1; j <= 4; j++) {
-        for (var i = 1; i <= 8; i++) {
+    for (var j = 0; j < 4; j++) {
+        for (var i = 0; i < 8; i++) {
+            var sampNo = 0;
             if (MC7000.PADModeSampler[j]) {
-                if (engine.getValue("[Sampler"+i+"]", "track_loaded") === 1) {
-                    if (engine.getValue("[Sampler"+i+"]", "play") === 0) {
-                        midi.sendShortMsg(0x94 + j - 1, 0x14 + i - 1, MC7000.padColor.samplerloaded);
+                if (MC7000.SamplerQty === 16) {
+                    (j >= 2) ? sampNo = (j - 2) * 8 + i + 1 : sampNo = j * 8 + i + 1;
+                } else if (MC7000.SamplerQty === 32) {
+                    sampNo = j * 8 + i + 1;
+                }
+                if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 1) {
+                    (engine.getValue("[Sampler"+sampNo+"]", "play") === 0) ? midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.samplerloaded) : midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.samplerplay);
+                } else if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 0) {
+                    midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.sampleroff);
+                }
+            }
+        }
+    }
+};
+
+// Velocity Sampler LED
+MC7000.VelSampLED = function() {
+    for (var j = 0; j < 4; j++) {
+        for (var i = 0; i < 8; i++) {
+            var sampNo = 0;
+            if (MC7000.PADModeVelSamp[j]) {
+                if (MC7000.SamplerQty === 16) {
+                    (j >= 2) ? sampNo = (j - 2) * 8 + i + 1: sampNo = j * 8 + i + 1;
+                } else {
+                    sampNo = j * 8 + i + 1;
+                }
+                if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 1) {
+                    var samplerIsNotPlaying = engine.getValue("[Sampler"+sampNo+"]", "play") === 0;
+                    if (samplerIsNotPlaying) {
+                        midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.velsamploaded); engine.setValue("[Sampler"+sampNo+"]", "pregain", 1);
                     } else {
-                        midi.sendShortMsg(0x94 + j - 1, 0x14 + i - 1, MC7000.padColor.samplerplay);
+                        midi.sendShortMsg(0x94 + j, 0x14 + i - 1, MC7000.padColor.velsampplay);
                     }
-                } else if (engine.getValue("[Sampler"+i+"]", "track_loaded") === 0) {
-                    midi.sendShortMsg(0x94 + j - 1, 0x14 + i - 1, MC7000.padColor.sampleroff);
+                } else if (engine.getValue("[Sampler"+sampNo+"]", "track_loaded") === 0) {
+                    midi.sendShortMsg(0x94 + j, 0x14 + i, MC7000.padColor.velsampoff);
                 }
             }
         }
     }
 };
 
+// Pitch LED when loading a track with already existing hotcues for pitch
+MC7000.PitchLED = function(value, group) {
```

Comment:
I do not understand how this mode is supposed to work.

Reply:
The pitch play function is one of the more complex ones regarding usage and corresponding colors of the buttons. When starting up, it requires the hotcue color to see if hotcues already exist or if not to create one on the fly. After that the colors need to change to pitch color with the active one bright and the others dark.

Reply:
I guess like described here: https://support.serato.com/hc/en-us/articles/115003094007-Using-the-Pitch-Play-mode